packages/wp-story-editor/src/api/story.js

```diff
@@ -29,6 +29,52 @@ import apiFetch from '@wordpress/api-fetch';
 import { STORY_EMBED, STORY_FIELDS } from './constants';
 import { base64Encode } from './utils';
 
+// Add custom fields and prepare story response.
+const transformGetStoryResponse = (post) => {
+  const { _embedded: embedded = {}, _links: links = {} } = post;
+
+  post.author = {
+    id: embedded?.author?.[0].id || 0,
+    name: embedded?.author?.[0].name || '',
+  };
+
+  post.capabilities = {
+    hasPublishAction: Object.prototype.hasOwnProperty.call(
+      links,
+      'wp:action-publish'
+    ),
+    hasAssignAuthorAction: Object.prototype.hasOwnProperty.call(
+      links,
+      'wp:action-assign-author'
+    ),
+  };
+
+  post.lock_user = {
+    id: embedded?.['wp:lockuser']?.[0].id || 0,
+    name: embedded?.['wp:lockuser']?.[0].name || '',
+    avatar: embedded?.['wp:lockuser']?.[0].avatar_urls?.['96'] || '',
+  };
+
+  post.featured_media = {
+    id: embedded?.['wp:featuredmedia']?.[0].id || 0,
+    height: embedded?.['wp:featuredmedia']?.[0]?.media_details?.height || 0,
+    width: embedded?.['wp:featuredmedia']?.[0]?.media_details?.width || 0,
+    url: embedded?.['wp:featuredmedia']?.[0]?.source_url || '',
+  };
```

Comment:
This logic is repeated a lot. Worth it's own function. Something like 

```
function getEmbeddedMediaObject( embedded, field ) {
    return {
          id: embedded?.[field]?.[0].id || 0,
          height: embedded?.[field]?.[0]?.media_details?.height || 0,
          width: embedded?.[field]?.[0]?.media_details?.width || 0,
          url: embedded?.[field]?.[0]?.source_url || '',
  };
}
```

Reply:
IIRC we discussed this very recently and my take was that using optional chaining like here is better for readability.